src/include/catalog.h

```diff
@@ -0,0 +1,24 @@
+#include <vector>
+#include "table.h"
+#include <string>
+
+namespace emerald {
+    class Catalog {
+        private:
+          std::vector<Table*> TableList;
+        public:
+          Catalog();
+          void AddTable(DbFile* file, std::string name, std::string pkeyField);
+          void AddTable(DbFile* file, std::string name);
+          void AddTable(DbFile* file);
+          int get_table_id(std::string name);
+          TupleDesc* GetTupleDesc(int tableid);
```

Comment:
The `GetDbFile()` function is missing, please add it in.

Reply:
I do not see a GetDbFile there

Reply:
There actually is a GetDbFile() function there in Catalog.java (Lines 104 - 116). But I'll add it in for you, no worries.